mlflow/pipelines/steps/predict.py

```diff
@@ -129,21 +129,36 @@ def _run(self, output_directory):
             if spark:
                 _logger.info("Found active spark session")
             else:
-                spark = _create_local_spark_session_for_pipelines()
                 _logger.info("Creating new spark session")
+                spark = _create_local_spark_session_for_pipelines()
         except Exception as e:
             raise MlflowException(
                 message=(
-                    "Encountered an error while searching for an active Spark session to"
-                    " score dataset with spark UDF. Please create a Spark session and try again."
+                    "Encountered an error while getting or creating an active Spark session to"
+                    " score dataset with spark UDF."
                 ),
                 error_code=BAD_REQUEST,
             ) from e
-        if not spark:
+
+        # check if output location is already populated for non-delta output formats
+        output_format = self.step_config["output_format"]
+        output_location = self.step_config["output_location"]
+        output_populated = False
+        if self.write_mode == "default":
```

Comment:
There's a synonym for `default`: 
[ErrorIfExists](https://spark.apache.org/docs/1.4.0/api/java/org/apache/spark/sql/SaveMode.html#ErrorIfExists)

Also what do we do with [Ignore](https://spark.apache.org/docs/1.4.0/api/java/org/apache/spark/sql/SaveMode.html#Ignore)?
I think we should print a warning and exist with status 0 (success) in this case for Parquet.

Reply:
I just implemented `default` because I didn't want to re-implement how spark handles save mode options. But it seems like this write mode options enum is small and has been around for a while.

I can add "default" and its synonyms in this PR. I can also rename "write_mode" to "save_mode" to make it more consistent with the Spark API. For the `ignore` case, maybe I can add it in a follow up PR? I'm guessing that I would need to create a new custom result card for this case, all of scoring would be skipped.

Reply:
Actually, would you be comfortable not short circuiting "ignore"? To use this option, a user would need to open the spark SaveMode docs, read that doc, and deliberately set the SaveMode to "ignore" - so they should be aware of the consequences. We can just document that the MLP batch artifacts would be updated but the output location would not. We short-circuit "default" because we know an error will happen during the spark writing, which we can flag early. For "ignore", the spark writing will not have an error.

Reply:
SGTM!